src/sail_json_backend/json.ml

```diff
@@ -202,6 +204,24 @@ let add_assembly app_id p =
       Hashtbl.add assembly (string_of_id app_id) x
     end
 
+
+
+let json_of_registers () =
+  let regs = Hashtbl.fold (fun name regtype accum ->
+    (name, regtype) :: accum
+  ) registers [] in
+  let sorted_regs = List.sort (fun (_, t1) (_, t2) -> String.compare t1 t2) regs in
+  let json_regs = List.map (fun (name, regtype) ->
+    "\n  {\n" ^
+    "    \"name\": \"" ^ name ^ "\",\n" ^
+    "    \"type\": \"" ^ regtype ^ "\"\n  },"
+  ) sorted_regs in
+  let reg_list = String.concat "" json_regs in
+  let final_reg_list = if reg_list = "" then "" else String.sub reg_list 0 (String.length reg_list - 1) in
+  "[ " ^ final_reg_list ^ "\n]"
+    
+
+
```

Comment:
Two extra blank lines.
The convention I've used so far is just a single blank line between functions.